relay-event-normalization/src/trimming.rs

```diff
@@ -948,19 +948,23 @@ mod tests {
             platform: Annotated::new("a".repeat(1024 * 100)),
             ..Default::default()
         };
-        let spans = std::iter::repeat_with(|| Annotated::new(span.clone()))
+        let spans: Vec<_> = std::iter::repeat_with(|| Annotated::new(span.clone()))
             .take(10)
             .collect();
 
         let mut event = Annotated::new(Event {
-            spans: Annotated::new(spans),
+            spans: Annotated::new(spans.clone()),
             ..Default::default()
         });
 
         let mut processor = TrimmingProcessor::new();
         processor::process_value(&mut event, &mut processor, ProcessingState::root()).unwrap();
 
-        assert_eq!(event.0.unwrap().spans.0.unwrap().len(), 8);
+        let trimmed_spans = event.0.unwrap().spans.0.unwrap();
+        assert_eq!(trimmed_spans.len(), 8);
+
+        // The actual spans were not touched:
+        assert_eq!(trimmed_spans.as_slice(), &spans[0..8]);
```

Comment:
This assertion fails without the changes to `Span`.

Reply:
Nice!